tests/test_RBAC.py

```diff
@@ -0,0 +1,173 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+import json
+
+import pytest
+from ocean_lib.common.agreements.service_types import ServiceTypes
+from ocean_lib.models.data_token import DataToken
+
+from ocean_provider.constants import BaseURLs
+from ocean_provider.exceptions import RequestNotFound
+from ocean_provider.utils.accounts import generate_auth_token
+from ocean_provider.validation.algo import build_stage_output_dict
+from ocean_provider.validation.requests import RBACValidator
+from tests.helpers.compute_helpers import build_and_send_ddo_with_compute_service
+from tests.test_helpers import (
+    get_dataset_ddo_with_access_service,
+    mint_tokens_and_wait,
+    send_order,
+)
+
+
+def test_null_validator():
+    with pytest.raises(RequestNotFound):
+        RBACValidator(request=None)
+
+
+encrypt_endpoint = BaseURLs.ASSETS_URL + "/encrypt"
+
+
+def test_encrypt_request_payload():
+    document = [
+        {
+            "url": "http://localhost:8030" + encrypt_endpoint,
+            "index": 0,
+            "checksum": "foo_checksum",
+            "contentLength": "4535431",
+            "contentType": "text/csv",
+            "encoding": "UTF-8",
+            "compression": "zip",
+        }
+    ]
+    req = {"document": json.dumps(document[0])}
+    validator = RBACValidator(request_name="EncryptRequest", request=req)
+    payload = validator.build_payload()
+    assert validator.request == req
+    assert payload
```

Comment:
no need to assert the payload if you are asserting its content. Instead of checking against the validator.action (component, credentials...), can we test the values themselves?